lib/index.ts

```diff
@@ -93,17 +90,14 @@ export function filterProcessList(rootPid: number, processList: IProcessInfo[],
   return children.reduce((prev, current) => prev.concat(current), [rootProcess]);
 }
 
-function getRawProcessList(
-  pid: number,
-  queue: RequestQueue,
-  callback: (processList: IProcessInfo[] | IProcessTreeNode | undefined) => void,
-  filter: (pid: number, processList: IProcessInfo[] | undefined, maxDepth: number) => IProcessInfo[] | IProcessTreeNode | undefined,
+function getRawProcessList<T>(
```

Comment:
`T extends IProcessInfo[] | IProcessTreeNode | undefined`?

Reply:
What would be a reason to add that constraint? The function is more strongly typed without it. It guarantees that this function is oblivious to the actual transformation and only handles the queue/callback management.

Reply:
Good point, been a while since I was in this codebase and thought it only worked on that type, but that's not true.